Show warning message when adding a dependency listed in the catalogs config

parksb opened the pull request:
![image](https://github.com/user-attachments/assets/d9d3ec7a-b0d0-4739-8823-f7f9ac32a5f6)

When adding a dependency using `yarn add` command, the catalog protocol may not be used even if the dependency is already listed in the catalogs config. This PR adds a warning message to inform the user that the dependency is included in the catalogs config in such cases.

For now, it only displays a gentle warning message, but in the future, it could take a more aggressive approach (such as throwing an error, enforcing the use of the catalog protocol, ...) through a configurable options.

jwoo0122 commented:
What about adding an option for using catalog as a default protocol when there's no explicit version in add command? Like

```yaml
catalog:
  defaultCatalog: stable
```
```bash
yarn add es-toolkit # will install es-toolkit@catalog:stable
```

and without the `defaultCatalog` it behaves as now.

parksb commented:
As you suggested, I added a separate option to configure the default alias group.

I also split the configuration into two parts: `options` and `list`. This allows dependencies with names like `default` or `root` to be listed in the catalog. In addition, the default field is now an array, which enables specifying multiple alias groups. Please see below for a concrete example of how it works.

You can set a default catalogs using the `default` option. 

```yaml
catalogs:
  options:
    default: ["beta"]
  list:
    beta:
      react: 19.0.0
      react-dom: 19.0.0
    legacy:
      react: 17.0.2
      react-dom: 17.0.2
```

```sh
yarn add react # Same as `yarn add react@catalog:beta`
```

If you want to use the root catalogs as the default, just set the `default` option to `root`.

```yaml
catalogs:
  options:
    default: ["root"]
  list:
    react: 19.0.0
    react-dom: 19.0.0
```

```sh
yarn add react # Same as `yarn add react@catalog:`
```

You can also set multiple default catalogs. In this case, the priority is determined by the order specified in the list.

```yaml
catalogs:
  options:
    default: ["stable", "beta", "root"]
  list:
    es-toolkit: 1.34.1
    beta:
      react: 19.0.0
      react-dom: 19.0.0
    stable:
      typescript: 5.1.6
```

```sh
yarn add typescript # Same as `yarn add typescript@catalog:stable`
yarn add react # Same as `yarn add react@catalog:beta`
yarn add es-toolkit # Same as `yarn add react@catalog:`
```

parksb commented:
I extended the `default` option to accept a string type. If `max` is passed, it will use the most frequently used alias group in package.json as the default group.

```yaml
catalogs:
  options:
    default: max
  list:
    beta:
      react: 18.0.0
      lodash: 3.0.0
      next: 12.0.0
      es-toolkit: 1.32.0
    stable:
      react: 17.0.0
      lodash: 2.0.0
      next: 11.0.0     
      es-toolkit: 1.31.0
```

```json
{
  "name": "test-workspace",
  "version": "1.0.0",
  "dependencies": {
    "react": "catalog:stable",
    "lodash": "catalog:stable",
    "next": "catalog:beta"
  }
}
```

```sh
yarn add es-toolkit # Same as `yarn add es-toolkit@catalog:stable`
```

jwoo0122 commented on README.md:
````diff
@@ -95,6 +98,66 @@ catalogs:
 }
 ```
 
+#### Default Catalogs
+
+You can configure default catalogs using the `default` option, allowing you to run `yarn add` without specifying a catalogs name.
+
````
We should mention "max" option.
